python/python/Ice/InitializationData.py

```diff
@@ -1,44 +1,55 @@
 # Copyright (c) ZeroC, Inc.
 
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from IcePy import Connection
+from .BatchRequestInterceptor import BatchRequestInterceptor
+from .EventLoopAdapter import EventLoopAdapter
+from .Logger import Logger
+from .Properties import Properties
+from .UserException import UserException
+from .Value import Value
+
+@dataclass
 class InitializationData:
     """
     The attributes of this class are used to initialize a new communicator instance.
 
     Attributes
     ----------
-    properties : Ice.Properties
+    properties : Ice.Properties | None
         You can use the Ice.createProperties function to create a new property set.
-    logger : Ice.Logger
+    logger : Ice.Logger | None
         The logger to use for the communicator.
-    threadStart : callable
+    threadStart : Callable[[], None] | None
         A callable that is invoked for each new Ice thread that is started.
-    threadStop : callable
+    threadStop : Callable[[], None] | None
         A callable that is invoked when an Ice thread is stopped.
-    executor : callable
+    executor : Callable[[Callable[[], None], Connection], None] | None
         A callable that is invoked when Ice needs to execute an activity. The callable
         receives two arguments: a callable and an Ice.Connection object. The executor must
         eventually invoke the callable with no arguments.
-    batchRequestInterceptor : callable
+    batchRequestInterceptor : BatchRequestInterceptor | None
```

Comment:
Current, this batchRequestInterceptor:

```
"batch request interceptor must either be a callable or an object with an 'enqueue' method"
```

I don't think we should use this "or" pattern. Just pick one. In this case, I would get rid of the  ABC and use a Callable.